awx/ui_next/src/screens/Template/shared/SurveyList.jsx

```diff
@@ -1,129 +1,75 @@
-import { getQSConfig } from '@util/qs';
 import AlertModal from '@components/AlertModal';
 import SurveyListItem from './SurveyListItem';
 import SurveyToolbar from './SurveyToolbar';
 
-const QS_CONFIG = getQSConfig('survey', {
-  page: 1,
-});
-
-function SurveyList({ template, i18n }) {
+function SurveyList({
+  isLoading,
+  survey,
+  surveyEnabled,
+  toggleSurvey,
+  updateSurvey,
+  deleteSurvey,
+  i18n,
+}) {
+  const questions = survey?.spec || [];
   const [selected, setSelected] = useState([]);
-  const [surveyEnabled, setSurveyEnabled] = useState(template.survey_enabled);
-  const [showToggleError, setShowToggleError] = useState(false);
   const [isDeleteModalOpen, setIsDeleteModalOpen] = useState(false);
 
-  const {
-    result: { questions, name, description },
-    error: contentError,
-    isLoading,
-    request: fetchSurvey,
-  } = useRequest(
-    useCallback(async () => {
-      const {
-        data: { spec = [], description: surveyDescription, name: surveyName },
-      } = await JobTemplatesAPI.readSurvey(template.id);
-      return {
-        questions: spec.map((s, index) => ({ ...s, id: index })),
-        description: surveyDescription,
-        name: surveyName,
-      };
-    }, [template.id]),
-    { questions: [], name: '', description: '' }
-  );
-
-  useEffect(() => {
-    fetchSurvey();
-  }, [fetchSurvey]);
-
   const isAllSelected =
     selected.length === questions?.length && selected.length > 0;
 
-  const {
-    isLoading: isDeleteLoading,
-    deleteItems: deleteQuestions,
-    deletionError,
-    clearDeletionError,
-  } = useDeleteItems(
-    useCallback(async () => {
-      if (isAllSelected) {
-        return JobTemplatesAPI.destroySurvey(template.id);
-      }
-      const surveyQuestions = [];
-      questions.forEach(q => {
-        if (!selected.some(s => s.id === q.id)) {
-          surveyQuestions.push(q);
-        }
-      });
-      return JobTemplatesAPI.updateSurvey(template.id, {
-        name,
-        description,
-        spec: surveyQuestions,
-      });
-      // eslint-disable-next-line react-hooks/exhaustive-deps
-    }, [selected]),
-    {
-      qsConfig: QS_CONFIG,
-      fetchItems: fetchSurvey,
-    }
-  );
-  const {
-    isToggleLoading,
-    error: toggleError,
-    request: toggleSurvey,
-  } = useRequest(
-    useCallback(async () => {
-      await JobTemplatesAPI.update(template.id, {
-        survey_enabled: !surveyEnabled,
-      });
-      return setSurveyEnabled(!surveyEnabled);
-    }, [template, surveyEnabled]),
-    template.survey_enabled
-  );
-
-  useEffect(() => {
-    if (toggleError) {
-      setShowToggleError(true);
-    }
-  }, [toggleError]);
-
   const handleSelectAll = isSelected => {
     setSelected(isSelected ? [...questions] : []);
   };
 
   const handleSelect = item => {
-    if (selected.some(s => s.id === item.id)) {
-      setSelected(selected.filter(s => s.id !== item.id));
+    if (selected.some(q => q.variable === item.variable)) {
```

Comment:
We feel pretty confident that variable is unique?

Reply:
The `variable` field should be unique - you can't have two questions with the same `variable` value in a survey